core/src/banking_stage.rs

```diff
@@ -360,11 +364,21 @@ impl BankingStage {
         socket: &std::net::UdpSocket,
         tpu_forwards: &std::net::SocketAddr,
         unprocessed_packets: &UnprocessedPackets,
+        data_budget: &DataBudget,
     ) -> std::io::Result<()> {
         let packets = Self::filter_valid_packets_for_forwarding(unprocessed_packets.iter());
         inc_new_counter_info!("banking_stage-forwarded_packets", packets.len());
+        const INTERVAL_MS: u64 = 100;
+        const MAX_BYTES_PER_SECOND: usize = 10_000 * 1200;
+        const MAX_BYTES_PER_INTERVAL: usize = MAX_BYTES_PER_SECOND * INTERVAL_MS as usize / 1000;
+        const MAX_BYTES_BUDGET: usize = MAX_BYTES_PER_INTERVAL * 5;
+        data_budget.update(INTERVAL_MS, |bytes| {
+            std::cmp::min(bytes + MAX_BYTES_PER_INTERVAL, MAX_BYTES_BUDGET)
+        });
         for p in packets {
-            socket.send_to(&p.data[..p.meta.size], &tpu_forwards)?;
+            if data_budget.take(p.meta.size) {
```

Comment:
I think I'd like to rebase https://github.com/solana-labs/solana/pull/19611 off of this PR and apply the data budget before sorting by weight fyi. Could be worth applying this data budget as part of the `filter_valid_packets_for_forwarding` filter with that in mind.

Reply:
ah nvm, missed that this change was already merged into v1.6 as https://github.com/solana-labs/solana/pull/19834

Reply:
Yea, I was trying to keep the branches consistent.

Reply:
We can make a follow-up change that applies on top with the changes you suggest.